core/templates/pages/profile-page/profile-page.component.spec.ts

```diff
@@ -16,118 +16,426 @@
+    fetchProfileDataAsync(): Promise<Object> {
+      return Promise.resolve(profileData);
     }
-  };
 
-  beforeEach(angular.mock.module('oppia', $provide => {
-    let ugs = new UpgradedServices();
-    for (let [key, value] of Object.entries(ugs.getUpgradedServices())) {
-      $provide.value(key, value);
+    subscribeAsync(username: string): Promise<void> {
+      return Promise.resolve();
     }
-  }));
 
-  beforeEach(angular.mock.module('oppia', function($provide) {
-    $provide.value('UrlService', {
-      getUsernameFromProfileUrl: () => 'username1'
-    });
-    $provide.value('WindowRef', windowRefMock);
+    unsubscribeAsync(username: string): Promise<void> {
+      return Promise.resolve();
+    }
+  }
+
+  class MockUrlService {
+    getUserFromProfileUrl(): string {
+      return 'username1';
+    }
+  }
+
+  beforeEach(waitForAsync(() => {
+    TestBed.configureTestingModule({
+      imports: [
+        HttpClientTestingModule,
+        MatCardModule
+      ],
+      declarations: [
+        MockTranslatePipe,
+        ProfilePageComponent,
+        TruncatePipe
+      ],
+      providers: [
+        {
+          provide: ProfilePageBackendApiService,
+          useClass: MockProfilePageBackendApiService
+        },
+        {
+          provide: UrlService,
+          useClass: MockUrlService
+        },
+        {
+          provide: WindowRef,
+          useClass: MockWindowRef
+        }
+      ],
+      schemas: [NO_ERRORS_SCHEMA]
+    }).compileComponents();
   }));
 
   beforeEach(() => {
-    TestBed.configureTestingModule({
-      imports: [HttpClientTestingModule],
-      providers: [ProfilePageBackendApiService]
-    });
-    OppiaAngularRootComponent.profilePageBackendApiService = (
-      TestBed.get(ProfilePageBackendApiService)
-    );
-    OppiaAngularRootComponent.ratingComputationService = (
-      TestBed.get(RatingComputationService));
+    fixture = TestBed.createComponent(ProfilePageComponent);
+    componentInstance = fixture.componentInstance;
+    userService = TestBed.inject(UserService);
+    userService = (TestBed.inject(UserService) as unknown) as
+      jasmine.SpyObj<UserService>;
+    csrfTokenService = TestBed.inject(CsrfTokenService);
+    csrfTokenService = (csrfTokenService as unknown) as
+      jasmine.SpyObj<CsrfTokenService>;
+    dateTimeFormatService = TestBed.inject(DateTimeFormatService);
+    dateTimeFormatService = (dateTimeFormatService as unknown) as
+      jasmine.SpyObj<DateTimeFormatService>;
+    loaderService = (TestBed.inject(LoaderService) as unknown) as
+      jasmine.SpyObj<LoaderService>;
+    loggerService = TestBed.inject(LoggerService);
+    loggerService = (loggerService as unknown) as LoggerService;
+    mockWindowRef = (TestBed.inject(WindowRef) as unknown) as MockWindowRef;
+    profilePageBackendApiService = (
+      TestBed.inject(ProfilePageBackendApiService) as unknown) as
+      jasmine.SpyObj<ProfilePageBackendApiService>;
```

Comment:
Why so many `as unknown` here? Also don't redefine the variable, either use a different name or put the `TestBed.inject(…)` part directly in the second assigment.

Reply:
fixed